test/parallel/test-crypto-subtle-cross-realm.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const { subtle } = crypto;
+const vm = require('vm');
+
+// Test with same-realm ArrayBuffer
+{
+  const samerealmData = new Uint8Array([1, 2, 3, 4]).buffer;
+
+  subtle.digest('SHA-256', samerealmData)
+    .then((result) => {
+      assert(result instanceof ArrayBuffer);
+      assert.strictEqual(result.byteLength, 32); // SHA-256 is 32 bytes
+    })
+    .catch(common.mustNotCall());
+}
+
+// Test with cross-realm ArrayBuffer
+{
+  const context = vm.createContext({});
+  const crossrealmUint8Array = vm.runInContext('new Uint8Array([1, 2, 3, 4])', context);
+  const crossrealmBuffer = crossrealmUint8Array.buffer;
+
+  // Verify it's truly cross-realm
+  assert.notStrictEqual(
+    Object.getPrototypeOf(crossrealmBuffer),
+    ArrayBuffer.prototype
+  );
+
+  // This should still work, since we're checking structural type
+  subtle.digest('SHA-256', crossrealmBuffer)
+    .then((result) => {
+      assert(result instanceof ArrayBuffer);
+      assert.strictEqual(result.byteLength, 32); // SHA-256 is 32 bytes
+    })
+    .catch(common.mustNotCall('Should not reject cross-realm ArrayBuffer'));
```

Comment:
Likewise here. wrap the then handler in `common.mustCall(...)` and drop the catch handler.